addons/binding/org.openhab.binding.zoneminder/src/main/java/org/openhab/binding/zoneminder/internal/state/MonitorThingState.java

```diff
@@ -0,0 +1,537 @@
+        super(subscriber);
+
+    }
+
+    @Override
+    public GenericChannelState createChannelSubscription(ChannelUID channelUID) {
+        GenericChannelState channelState = null;
+        try {
+            switch (channelUID.getId()) {
+                case ZoneMinderConstants.CHANNEL_MONITOR_DETAILED_STATUS:
+                case ZoneMinderConstants.CHANNEL_MONITOR_EVENT_CAUSE:
+                case ZoneMinderConstants.CHANNEL_MONITOR_FUNCTION:
+                    channelState = createSubscriptionStringType(channelUID);
+                    break;
+
+                case ZoneMinderConstants.CHANNEL_MONITOR_STILL_IMAGE:
+                    channelState = createSubscriptionRawType(channelUID);
+                    break;
+                default:
+                    channelState = createSubscriptionOnOffType(channelUID);
+                    break;
+            }
+
+        } catch (Exception ex) {
+            logger.error("{}: context='subscribe' - Exception occurred when subscribing to channel '{}'", "<UNKNOWN>",
+                    channelUID.getId());
+        }
+        return channelState;
+    }
+
+    public boolean isDirty() {
+        return isDirty;
+    }
+
+    public boolean isAlarmed() {
+        State stateEnabled = getChannelStateHandler(ZoneMinderConstants.CHANNEL_MONITOR_EVENT_STATE).getState();
+        return ((alarmedDetailedState.get() || (stateEnabled == OnOffType.ON)) ? true : false);
+
+    }
+
+    private boolean isEnabled() {
+        State stateEnabled = getChannelStateHandler(ZoneMinderConstants.CHANNEL_MONITOR_ENABLED).getState();
+        if (stateEnabled == OnOffType.ON) {
+            return true;
+        }
+        return false;
+    }
+
+    public ZoneMinderMonitorFunctionEnum getMonitorFunction() {
+        State stateFunction = getChannelStateHandler(ZoneMinderConstants.CHANNEL_MONITOR_FUNCTION).getState();
+        return ZoneMinderMonitorFunctionEnum.getEnum(stateFunction.toString());
+    }
+
+    public String getMonitorEventCause() {
+        State stateEventCause = getChannelStateHandler(ZoneMinderConstants.CHANNEL_MONITOR_EVENT_CAUSE).getState();
+        return stateEventCause.toString();
+    }
+
+    public boolean getMonitorEventMotion() {
+        State stateEventMotion = getChannelStateHandler(ZoneMinderConstants.CHANNEL_MONITOR_MOTION_EVENT).getState();
+        return false;
+
+    }
+
+    public ZoneMinderMonitorStatusEnum getMonitorDetailedStatus() {
+        State stateStatus = getChannelStateHandler(ZoneMinderConstants.CHANNEL_MONITOR_DETAILED_STATUS).getState();
+        ZoneMinderMonitorStatusEnum result = ZoneMinderMonitorStatusEnum.UNKNOWN;
+
+        if (stateStatus == UnDefType.NULL || stateStatus == UnDefType.UNDEF) {
+            return ZoneMinderMonitorStatusEnum.UNKNOWN;
+        }
+
+        try {
+            result = ZoneMinderMonitorStatusEnum.getEnumFromName(stateStatus.toString());
+        } catch (Exception ex) {
```

Comment:
Please catch specific exceptions instead of catching all exceptions. That way it is more clear what exceptions need to be handled and how this is done.